runtime/lua/vim/diagnostic.lua

```diff
@@ -751,6 +752,18 @@ function M.get_namespaces()
   return vim.deepcopy(all_namespaces)
 end
 
+---@class Diagnostic
+---@field buffer number
+---@field lnum number 0-indexed
+---@field end_lnum nil|number 0-indexed
+---@field col number 0-indexed
+---@field end_col nil|number 0-indexed
+---@field severity DiagnosticSeverity
+---@field message string
+---@field source nil|string
```

Comment:
Isn't it better to use `?`:

```suggestion
---@field source? string
```

Reply:
I don't think that works with `gen_vimdoc.py`. Also, we've standardized `(string|nil)` everywhere else, and we should be consistent.

Reply:
There is one usage in `_editor.lua` and it appears to work. I definitely think `?` is more idiomatic and communicates "this argument doesn't need to be provided" as opposed to "this argument can be `nil`".

Reply:
> There is one usage in _editor.lua and it appears to work.

That's good to know.

Nevertheless, I strongly prefer consistency. If we switch, we should switch everywhere.